### ISSUE TITLE
Audio Devices Other Than Default Not Shown on Linux

### ISSUE THREAD

**Issue by yoshiyoshyosh** [Yellow-Dog-Man/Resonite-Issues#1748]
### Describe the bug?

After the advent of the new settings update, audio input devices other than "Default", "Steam Voice", and "Audio Loopback" are unavailable on Linux Native. No audio output devices are shown at all. This behavior is not present on Proton.

### To Reproduce

1. Launch game on Linux Native with custom audio devices
2. Try to use an audio device other than default for output/input
3. Realize that you can't, and be left sad

### Expected behavior

All available audio devices are shown.

### Screenshots

Screenshot of the audio input device screen along with my own audio devices in `pavucontrol`. Said audio devices do not show up in the setting:
![Screenshot of the audio input device screen along with my own audio devices in `pavucontrol`. Said audio devices do not show up in the setting.](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/36683489/bb5c71b0-8108-4883-b61d-37671d1722e1)

However, I am able to use these devices from the "Stream Audio" tab perfectly fine:
![2024-04-17_22-57-20](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/36683489/c51489aa-9fc8-44d6-bfe6-5773d2aca4ff)

### Resonite Version Number

2024.4.17.1420

### What Platforms does this occur on?

Linux

### Log Files

https://gist.github.com/yoshiyoshyosh/ddda0b446fe32c0cfb6d4d0d5a71a322

### Additional Context

My audio devices are added through pipewire's interface, as well as using a `pipewire-pulse` server, which should ensure the most compatibility.

This part of the log looks promising:
```
System.NotImplementedException: The method or operation is not implemented.
  at Hardware.Info.Windows.HardwareInfoRetrieval..ctor (System.Nullable`1[T] enumerationOptionsTimeout) [0x00032] in <74e30ffee6f94935b84f514c5a11c52f>:0 
  at Hardware.Info.HardwareInfo..ctor (System.Boolean useAsteriskInWMI, System.Nullable`1[T] timeoutInWMI) [0x000b7] in <74e30ffee6f94935b84f514c5a11c52f>:0 
  at SkyFrost.Base.UID.Compute () [0x00014] in <d4ec01a9e3164455945b728e633ac1cc>:0 
  at FrooxEngine.Engine.<InitializeSkyFrost>b__340_3 () [0x0003c] in <7c167c5ce3d1450585afe1c3de94f57c>:0 
```

### Reporters

yosh

**Comment by Frooxius** [comment]
This is not a bug, we do not support working with individual audio devices on Linux.

I don't believe this is something we even supported before the new Settings UI, were you able to choose individual audio devices on Linux before?

**Comment by yoshiyoshyosh** [comment]
> This is not a bug, we do not support working with individual audio devices on Linux. I don't believe this is something we even supported before the new Settings UI

Interesting!

> were you able to choose individual audio devices on Linux before?

Yes, I was able to choose individual audio devices before the settings update, which is why I filed this thinking it was a bug. The logs ~appear to indicate~ that the Unity Audio API is what was/is still able to be able to get the audio devices before the update and in the "Stream Audio" tab, but since you guys of course want to get off Unity, then... I don't know.

**Comment by shiftyscales** [comment]
Is this issue still relevant even when running the Windows build through Proton,@yoshiyoshyosh? If not, this can probably be closed for now as the native Linux client has been deprecated indefinitely.

**Comment by yoshiyoshyosh** [comment]
> Is this issue still relevant even when running the Windows build through Proton,[@yoshiyoshyosh](https://github.com/yoshiyoshyosh)? If not, this can probably be closed for now as the native Linux client has been deprecated indefinitely.

no, though the issue will most likely appear again once linux comes back online

**Comment by shiftyscales** [comment]
Closing for now then, thank you. Please re-open, or make a new issue if this becomes relevant again.
